packages/cli/src/decorators/registerController.ts

```diff
@@ -81,9 +82,12 @@ const authFreeRoutes: string[] = [];
 export const canSkipAuth = (method: string, path: string): boolean =>
 	authFreeRoutes.includes(`${method.toLowerCase()} ${path}`);
 
-export const registerController = (app: Application, config: Config, cObj: object) => {
-	const controller = cObj as Controller;
-	const controllerClass = controller.constructor;
+export const registerController = (
+	app: Application,
+	config: Config,
+	controllerClass: Class<object>,
+) => {
+	const controller = Container.get(controllerClass as Class<Controller>);
```

Comment:
```suggestion
export const registerController = (
	app: Application,
	config: Config,
	controllerClass: Class<Controller>,
) => {
	const controller = Container.get(controllerClass);
```

Reply:
I tried that first, but then had an error in Server.ts, and had to revert 
![image](https://github.com/n8n-io/n8n/assets/196144/cf0823de-58c8-4035-b890-a610dffe7a01)